### PR TITLE
Simplify Circle Ci config

### PR DESCRIPTION
**Pull request by Luthaf**
- Remove un-used files
- Remove duplication in config.yml
- Use a single point of knowledge (requirements.txt) for all Python dependencies

### CODE REVIEW

**Comment by Luthaf** [comment]
This is a first step toward #30.

**Comment by Luthaf** [comment]
@agoscinski, I think I discussed this with you: pip does not support conda style yaml files, and since some dependencies of rascal only exist on PyPi, I went with a pip-style requirements.txt file. This should still be usable with conda environments, using conda's pip.

**Comment by Luthaf** [comment]
Also ping @gibertif, since you wrote the circle Ci config initially.

**Comment by Luthaf** [comment]
Also, do we need to test on all version of GCC and Clang? We could just use the oldest and newest version of these (gcc 5 & 9, clang 4 & 9), thus cutting on CI run time.

File: README.rst [hunk 1]
```diff
@@ -12,13 +12,13 @@ collection of tools to this end.
 
 librascal is currently considered a standalone code. However, we aim to
 provide enough flexibility to interface it with other codes such as
-LAMMPS and PLUMED-2.0.  It can be used as a C++ library as well as a
-python module.  To be able to call it from python, we have used the
+LAMMPS and PLUMED-2.0. It can be used as a C++ library as well as a
```
Comment (max-veit) [inline README.rst]: French spacing...? In all seriousness, though, the double space makes it easier to read in code; this is of course ignored by the typesetting to give the "modern" single-space.

Comment (Luthaf) [reply README.rst]: No, I have [proselint](https://github.com/amperser/proselint) configured locally, and it was complaining :smile: 

I though this was standard in English too, but I can change it back if you want.

Comment (Luthaf) [reply README.rst]: The exact message was "Inconsistent spacing after period (1 vs. 2 spaces)." so I guess both are accepted in English, and proselint just want me to be more consistent :smile:

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+numpy
+scipy
+# for documentation
+sphinx==2.1.2
+breathe==4.13.1
+sphinx_rtd_theme
+nbsphinx
+# Developers tools
+jupyter
+cpplint
```
Comment (agoscinski) [inline requirements.txt]: I added some libraries, not essential for the tests but for running the tutorials and developer tools

Comment (max-veit) [reply requirements.txt]: I wouldn't call running the tutorials a "developer tool"

**Pull request merged**